pkg/alertmanager/alertstore/store.go

```diff
@@ -73,7 +73,7 @@ type AlertStore interface {
 }
 
 // NewAlertStore returns a alertmanager store backend client based on the provided cfg.
-func NewAlertStore(ctx context.Context, cfg Config, cfgProvider bucket.TenantConfigProvider, logger log.Logger, reg prometheus.Registerer) (AlertStore, error) {
+func NewAlertStore(ctx context.Context, cfg Config, cfgProvider bucket.TenantConfigProvider, fetchGrafanaCfg bool, logger log.Logger, reg prometheus.Registerer) (AlertStore, error) {
```

Comment:
I'd pass BucketAlertStoreConfig in here too.

Reply:
I decided to pass a boolean here to avoid importing `bucketclient` in package `mimir` 😬 
I'll import it then!